library/src/main/java/com/novoda/downloadmanager/StorageCapacityReader.java

```diff
@@ -0,0 +1,20 @@
+package com.novoda.downloadmanager;
+
+import android.os.Build;
+import android.os.StatFs;
+
+final class StorageCapacityReader {
+
+    private StorageCapacityReader() {
+        // Uses static factory method.
+    }
+
+    static long storageCapacityInBytes(StatFs statFs) {
```

Comment:
😂 let me try again.

`external` is not `sd` cards, let's get that cleared up first 😄 

`external` -> publicly available space on the device, not guaranteed to be an sd card.
`internal` -> root / app only access.

When looking for the `external` size it does not differentiate between the `primary` and `secondary` storage, so it will include both. 

To be quite honest, this is the first time I have spent any great deal of time looking at it. Perhaps @Hutch4 would be best place to explain this. He can also verify if I am barking up the wrong tree 😄